homework/index.js

```diff
@@ -29,18 +14,99 @@
+    document.getElementById('Description:').innerText = repository.description;
+    document.getElementById('Forks:').innerText = repository.forks;
+    document.getElementById('Updated:').innerText = new Date(
+      repository.updated_at,
+    ).toLocaleDateString();
+  }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function createContributorsSide(contributorContainer, contributors) {
+    contributorContainer.innerHTML = '';
+    createAndAppend('p', contributorContainer, { text: 'Contributions: ' });
+    contributors.forEach(contributor => {
+      const contributorInfo = createAndAppend('a', contributorContainer, {
+        href: contributor.html_url,
+        target: '_blank',
+      });
+      const contributorDiv = createAndAppend('div', contributorInfo, {
+        class: 'contributor',
+      });
+      createAndAppend('img', contributorDiv, { src: contributor.avatar_url });
+      createAndAppend('p', contributorDiv, { text: contributor.login });
+      createAndAppend('div', contributorDiv, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
     });
   }
 
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('div', root, { class: 'header' });
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+
+    fetch(url)
+      .then(Response => Response.json())
+      .then(repositories => {
+        const container = createAndAppend('div', root, { id: 'container' });
+        const repoContainer = createAndAppend('div', container, {
+          class: 'left-div',
+        });
+        const contributorContainer = createAndAppend('div', container, {
+          class: 'right-div',
+        });
+        const select = createAndAppend('select', header, { class: 'repo-selector' });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        repositories.forEach(repository => {
+          createAndAppend('option', select, { text: repository.name });
+        });
+        createRepositoryDescription(repoContainer, repositories[0]);
+        fetch(repositories[0].contributors_url)
+          .then(response => response.json())
+          .then(contributors => createContributorsSide(contributorContainer, contributors))
+          .catch(Error =>
+            createAndAppend('div', root, {
+              text: Error.message,
+              class: 'alert-error',
+            }),
+          );
+
+        select.addEventListener('change', () => {
+          const index = select.selectedIndex;
+          createRepositoryDescription(repoContainer, repositories[index]);
+          fetch(repositories[index].contributors_url)
```

Comment:
Do you notice how this code snippet is similar to one above where you are fetching contributors data for the first repo.

Event listener:
```
          fetch(repositories[index].contributors_url)
            .then(contributors => contributors.json())
            .then(contributors => {
              createContributorsSide(contributorContainer, contributors);
            })
            .catch(changeError =>
              createAndAppend('div', root, {
                text: changeError.message,
                class: 'alert-error',
              }),
            );
```
and fetching contributors data for first repo:
```
        fetch(repositories[0].contributors_url)
          .then(response => response.json())
          .then(contributors => createContributorsSide(contributorContainer, contributors))
          .catch(Error =>
            createAndAppend('div', root, {
              text: Error.message,
              class: 'alert-error',
            }),
          );
```

In such cases, it is better to move the repeated logic in its own function say `fetchContributorsData` and call this function wherever you want to use the functionality. Can you try to do it? 🙂